pkg/api/api.go

```diff
@@ -54,5 +57,20 @@ func NewClient(baseURL string, opts ...Option) *Client {
 // Do executes c.doFunc(), which in turn allows wrapping c.client.Do() and manipulating
 // the request behavior of the api client.
 func (c *Client) Do(req *http.Request) (*http.Response, error) {
-	return c.doFunc(c, req)
+	resp, err := c.doFunc(c, req)
+	if err != nil {
+		if errors.Is(err, syscall.ECONNREFUSED) {
```

Comment:
Will this match connection errors from `dns lookup` as well as errors connecting to `api.wakatime.com` servers?

We want to prevent using alternate DNS resolver when the connection fails to `api.wakatime.com`, and only use alternate DNS resolver when the `DNS lookup` fails.

Reply:
Let's check for `dial tcp: lookup` in the error message string before using alternate DNS resolver?

https://stackoverflow.com/a/67647035/1290627